queue/private.php

```diff
@@ -1,91 +1,107 @@
 <?php $root = realpath($_SERVER["DOCUMENT_ROOT"]) ?>
 <!DOCTYPE html>
-<html class='no-js theme-light' lang="">
+<html class="no-js theme-light" lang="" ng-app="queueApp">
 <head>
 	<title>Queue Test</title>
 	<?php require_once "$root/server/header.php" ?>
-	<link rel='stylesheet' href='/queue/queue_private.css'>
-	<meta http-equiv='refresh' content='600'/>
+	<link rel="stylesheet" href="/queue/queue.css">
+	<link rel="stylesheet" href="/queue/queue_private.css">
 </head>
-<body>
+<body ng-controller="QueuePrivateController">
 	<!--[if lt IE 8]>
 		<p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> to improve your experience.</p>
 	<![endif]-->
 	<?php
 		require_once "$root/components/nav.php";
 	?>
 
-	<div class='section flex box theme-light queue-details'>
-		<div class='flex box queue'>
-			<div class='theme-dark flex box date-wrap'>
-				<i class='material-icons no-select date-back'>keyboard_arrow_left</i>
-				<div class='date'></div>
-				<i class='material-icons no-select date-forward'>keyboard_arrow_right</i>
-			</div>
-			<div class='theme-dark flex box queue-header'>
-				<div class='wrap-left queue-position-wrap'>Pos.</div>
-				<div class='wrap-left queue-name-wrap'>Name</div>
-				<div class='wrap-right queue-time-wrap'>Time</div>
-			</div>
-			<div class='theme-white flex queue-table'></div>
-			<div class='flex queue-legend'>
-				Legend:
-				<div class='flex queue-tag on-time-tag'>On Time</div>
-				<div class='flex queue-tag late-tag'>Late</div>
-				<div class='flex queue-tag no-show-tag'>No Show</div>
-			</div>
-		</div>
-		<div class='flex details'>
-			<div class='theme-white flex details-id'>
-				<div class='theme-dark box details-id-header'>Details</div>
-				<div class='details-id-body'>
-					<div class='box details-id-attribute'>
-						<div class='details-id-attribute-label'>Name</div>
-						<div class='details-id-attribute-value details-name'></div>
-					</div>
-					<div class='box details-id-attribute'>
-						<div class='details-id-attribute-label'>Email</div>
-						<div class='details-id-attribute-value details-email'></div>
-					</div>
-					<div class='box details-id-attribute'>
-						<div class='details-id-attribute-label'>Phone</div>
-						<div class='details-id-attribute-value details-phone'></div>
-					</div>
-					<div class='box details-id-attribute'>
-						<div class='details-id-attribute-label'>Site</div>
-						<div class='details-id-attribute-value details-site-name'></div>
+	<!-- Header section -->
+	<?php
+		require_once "$root/queue/queue_header.php";
+	?>
+
+	<!-- Body Section -->
+	<div class="relative-wrapper clearfix">
+		<div class="fill-remaining d-flex flex-column flex-md-row">
```

Comment:
This part was a little weird. I made a [codepen](https://codepen.io/anon/pen/VzoWBX) to simplify it.

Basically I needed an absolutely positioned element in order to fill the remaining space, but those elements are positioned relative to the closes non-static parent. That's what the relative-wrapper is for. And then I had to do some clearfix tricks in order to trick the browser into thinking that the content filled the whole page if it actually takes less than the whole page.

Reply:
O_o